### ISSUE TITLE
Workflow-level option support

### ISSUE THREAD

**Issue by jkeifer** [stac-utils/stac-task#165]
The structure of `process` currently supports task-level options, or the fixed set of workflow-level options defined for `upload_options`. I've run into a circumstance where tasks run in a pipeline's workflows need to access a common option, and it would be super helpful when building the payload not to have to be concerned with what tasks in a given workflow might need to access that option, instead making it available to all tasks.

As it stands, all task-level options are passed into the task's `process` method as kwargs. Being able to leverage that same mechanism for workflow-level options would be fantastic: each task in a workflow would get a reference to the option via the kwargs whether it requires it or not. In the case of duplicate keys between a task's options and the workflow options, it seems that the value from the task options should be given precedence.

Could we add a new (optional) object to the `process` object, say `workflow_options`? Perhaps an example would be helpful:

```json
{
    "type": "FeatureCollection",
    "features": [
        ...
    ],
    "process": {
        "upload_options": {
            ...
        },
        "tasks": {
            "some-task": {
                "opt1": "bananas",
                "opt2": 42
            },
            "other-task": {}
        },
        "workflow_options": {
            "global_opt": ["a", "b", "c", "d"],
            "opt1": "apples"
        }
    }
}
```

If we passed this payload into a workflow with the tasks `some-task` and `other-task`, then the kwargs passed into the `process` method for each would be like the following:

```python
# some-task
kwargs = {
    "global_opt": ["a", "b", "c", "d"],
    "opt1": "bananas",
    "opt2": 42,
}

# other-task
kwargs = {
    "global_opt": ["a", "b", "c", "d"],
    "opt1": "apples",
}
```

Thoughts?

---

I considered the fact that stac workflow implementations like cirrus have `workflow` already defined in the `process` definition. Currently that is a string parameter that needs to equal the workflow name for which the payload is to be dispatched. I was thinking that it could make sense to support `workflow` as an object instead of a string, which would then have `name` and `options` keys:

```json
"workflow": {
    "name": "some-workflow",
    "options":  {
        "global_opt": ["a", "b", "c", "d"],
        "opt1": "apples"
    }
}
```

Except 1) `workflow` might not be supported by all stac workflow implementations (SWOOP maybe? I'm remembering a concern about having the workflow name in the payload, at least); 2) such a change would require updates on the cirrus side to support; and 3) `process` could be viewed, in some sense, as the `workflow` configuration object. With regard to the latter, it might then make sense not to name this `workflow_options` but just `options`. I think, though, given that the object is named `process` and that we already have `upload_options` that we need a more descriptive name to disambiguate it and make it clear what it is for. Other such names might be `global_task_options`, or just `global_options`, or something along those lines.

**Comment by jkeifer** [comment]
After discussion we have decided we are good proceed with the proposed plan. We do have reservations about the lack of clarity on the arguments passed into a task `process` method because of the use of `kwargs`, but that seems orthogonal to the concern here and we can address that with better parameter definition/validation in the future.

**Comment by pjhartzell** [comment]
Agreed on the way forward for now. One comment about the `process` block field names should we ever look at reworking this block and updating dependent libraries (`cirrus`, `swoop`). When I look at this:

```
{
    "type": "FeatureCollection",
    "features": [
        ...
    ],
    "process": {
        "upload_options": {
            ...
        },
        "tasks": {
            "some-task": {
                "opt1": "bananas",
                "opt2": 42
            },
            "other-task": {}
        },
        "workflow_options": {
            "global_opt": ["a", "b", "c", "d"],
            "opt1": "apples"
        }
    }
}
```


...it seems that the `tasks` field is really a `task_options` field. Renaming `tasks` to `task_options` would harmonize the three top-level field names such that their names match their intended use. Makes things clearer, IMO.

**Comment by pjhartzell** [comment]
Resolved by #167
